sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java

```diff
@@ -39,4 +45,28 @@
             new SharedTokenCacheCredential(null, "04b07795-8ddb-461a-bbee-02f9e1bf7b46",
                 identityClientOptions))));
     }
+
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        final AtomicReference<Throwable> cause = new AtomicReference<>();
+        StringBuilder errorMsg = new StringBuilder();
+        return Flux.fromIterable(credentials).flatMap(p -> p.getToken(request).onErrorResume(t -> {
```

Comment:
We have changed this to run in serial: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java#L46